_rules/text-node-contrast-afw4f7.md

```diff
@@ -0,0 +1,337 @@
+
+## Accessibility Support
+
+Different browsers have different levels of support for CSS. This can cause contrast issues in one browser that do not appear in another. Because of that, this rule can produce different results depending on the browser that is used.
+
+## Background
+
+- [Understanding Success Criterion 1.4.3: Contrast (Minimum)](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html)
+- [G18: Ensuring that a contrast ratio of at least 4.5:1 exists between text (and images of text) and background behind the text](https://www.w3.org/WAI/WCAG21/Techniques/general/G18)
+- [G145: Ensuring that a contrast ratio of at least 3:1 exists between text (and images of text) and background behind the text](https://www.w3.org/WAI/WCAG21/Techniques/general/G145)
+- [F83: Failure of Success Criterion 1.4.3 and 1.4.6 due to using background images that do not provide sufficient contrast with foreground text (or images of text)](https://www.w3.org/WAI/WCAG21/Techniques/failures/F83)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+Black [text node](https://dom.spec.whatwg.org/#text) on a white background.
+
+```html
+<p style="color: #333; background: #FFF;">
+	Some text in English
+</p>
+```
+
+#### Passed Example 2
+
+Black [text node](https://dom.spec.whatwg.org/#text) on a partially white gradient background.
+
+```html
+<p style="color: #333; background: linear-gradient(to right, #FFF, #00F);">
+	Some text in English
+</p>
+```
+
+#### Passed Example 3
+
+Light [text node](https://dom.spec.whatwg.org/#text) on a partially black background image.
+
+```html
+<p
+	style="color: #CCC; height:50px; padding-top:25px; background: #000 no-repeat -20px -20px url('../test-assets/black-hole.jpeg');"
+>
+	Black hole sun
+</p>
+```
+
+#### Passed Example 4
+
+[text node](https://dom.spec.whatwg.org/#text) that passes because of a text shadow.
+
+```html
+<p style="color: #000; background: #737373; text-shadow: white 0 0 3px">
+	Some text in English
+</p>
+```
+
+#### Passed Example 5
+
+18 point text with a 3:1 contrast.
+
+```html
+<p style="color: #000; font-size:18pt; background: #666;">
+	Some text in English
+</p>
+```
+
+#### Passed Example 6
+
+14 point bold text with a 3:1 contrast.
+
+```html
+<p style="color: #000; font-size:14pt; font-weight:700; background: #666;">
+	Some text in English
+</p>
+```
+
+#### Passed Example 7
+
+Decorative text with insufficient contrast.
+
+```html
+<p>Helvetica is a widely used sans-serif typeface developed in 1957 by Max Miedinger and Eduard Hoffmann.</p>
+<p style="font-family: helvetica; background: #EEE; color: #777;" aria-hidden="true">
+	The quick brown fox jumps over the lazy dog.
+</p>
+```
+
+#### Passed Example 8
+
+Text that does not express anything in human language.
+
+```html
+<p style="color: #000; background: #666;">
+	----=====++++++++___________***********%%%%%%%%%%%±±±±@@@@@@@@
```

Comment:
I kinda guess that the language is an input aspect because "human language" mean "of the page" rather than "any human language that exist(ed) on Earth".
In that case, I think that there should be an example with, say, a page in English and some poorly contrasted text in another language.

Also beware of "reversed spellings of ancient Chinese emperors" 🤣 
http://www.galactanet.com/comic/view.php?strip=284